src/dog/jax/dog.py

```diff
@@ -0,0 +1,241 @@
+"""
+    JAX implementation of the DoG/LDoG optimizers (Ivgi et al., 2023)
+"""
+
+import chex
+import jax
+import jax.lax as lax
+import jax.numpy as jnp
+import operator
+import optax
+from optax import ScalarOrSchedule
+from optax._src import base, combine, utils
+from optax._src.alias import _scale_by_learning_rate
+from typing import Optional, NamedTuple
+
+
+@jax.jit
+def _sum_params(params: base.Params) -> chex.Array:
+    """Compute sum(params)."""
+    sums = jax.tree_util.tree_map(jnp.sum, params)
+    return jax.tree_util.tree_reduce(operator.add, sums)
+
+
+@jax.jit
+def _params_squared_norm(params: base.Params, other: Optional[base.Params]=None) -> chex.Array:
```

Comment:
Have you considered using jax.numpy.linalg.norm instead?

Reply:
Done, thanks